src/qibolab/platforms/abstract.py

```diff
@@ -15,6 +18,68 @@
 )
 
 
+@dataclass
+class Qubit:
+    name: str
+    readout_frequency: int = 0
+    drive_frequency: int = 0
+    sweetspot: float = 0
+    peak_voltage: float = 0
+    pi_pulse_amplitude: float = 0
+    T1: int = 0
+    T2: int = 0
+    state0_voltage: int = 0
+    state1_voltage: int = 0
+    mean_gnd_states: complex = 0 + 0.0j
+    mean_exc_states: complex = 0 + 0.0j
+    resonator_polycoef_flux: List[float] = field(default_factory=list)
+
+
+@dataclass
+class ExecutionResults:
+    """Data strucutre to store output from `execute_pulse_sequence`."""
+
+    i: np.ndarray
+    q: np.ndarray
+    sample: np.ndarray
+
+    def msr(self):
+        """Computes MSR value."""
+        return np.sqrt(self.i**2 + self.q**2)
+
+    def phase(self):
+        """Computes phase value."""
+        return np.angle(self.i + 1j * self.q)
```

Comment:
These two could be `@property` so that we have symmetric interface with i and q. Now one has to do `result.i` but `result.msr()`. I would also consider the possibility of using capitals: `self.I, self.Q, self.MSR` which is closer to the physical notation.

Anyway this comment is just for appearance and anything would work for me, so feel free to ignore. Maybe @AleCandido has a stronger opinion.

Reply:
I agree they could be options, since they do not take arguments.

About capitals it is an option: it would violate Python naming convention (`snake_case` for variables and attributes, including functions and methods), but if it makes it more comfortable a few isolated violations are not necessarily bad.

I would still consider the option of dropping the class, but not necessarily: if you have a call that returns the object, with the class you have the method without importing, while functions you always need to import (but it is similar to importing NumPy functions to operate on arrays, so not unreasonable).